app/helpers/application_helper.rb

```diff
@@ -119,14 +119,12 @@ def deeds_for(options={})
   def show_prerender(prerender, locale) 
     begin
       prerenders = JSON.parse(prerender)
-      if prerenders.key? locale.to_s
-        # show prerender in specified locale
-        prerenders[locale.to_s]
-      else
+      unless render = prerenders[locale.to_s] # show prerender in specified locale
         # prerender doesn't have specified locale, show first fallback that prerender has
-        fallback = I18n.fallbacks[locale].select { |locale| prerenders.key? locale.to_s }.first
-        prerenders[fallback.to_s]
+        fallback = (I18n.fallbacks[locale].map(&:to_s) & prerenders.keys).first
+        render = prerenders[fallback]
       end
+      render
```

Comment:
This looks much better, but I'm a little concerned that the variable name `render` may be shadowing the method of the same name.  Can you rename `render` to something less suspicious?